src/raster/rasterband.rs

```diff
@@ -169,3 +183,85 @@ impl<'a> MajorObject for RasterBand<'a> {
 }
 
 impl<'a> Metadata for RasterBand<'a> {}
+
+/// # Read To Array
+/// A helper function that takes a raster band as input. It will
+/// read the rasterband into an ndarray.
+fn read_to_array<T: 'static + Copy + FromPrimitive>(rband: &RasterBand) -> Result<Array2<T>> {

```

Comment:
Since GDALDataType is now public, i think `T: GDALDataType`  could be enough here.